src/main/java/org/reactome/release/orthopairs/UniProtGeneNamesRetriever.java

```diff
@@ -0,0 +1,178 @@
+package org.reactome.release.orthopairs;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import uk.ac.ebi.kraken.interfaces.uniprot.Gene;
+import uk.ac.ebi.uniprot.dataservice.client.Client;
+import uk.ac.ebi.uniprot.dataservice.client.QueryResult;
+import uk.ac.ebi.uniprot.dataservice.client.ServiceFactory;
+import uk.ac.ebi.uniprot.dataservice.client.exception.ServiceException;
+import uk.ac.ebi.uniprot.dataservice.client.uniprot.UniProtComponent;
+import uk.ac.ebi.uniprot.dataservice.client.uniprot.UniProtQueryBuilder;
+import uk.ac.ebi.uniprot.dataservice.client.uniprot.UniProtService;
+import uk.ac.ebi.uniprot.dataservice.query.Query;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardOpenOption;
+import java.time.Duration;
+import java.util.*;
+
+public class UniProtGeneNamesRetriever {
+
+    private static final Logger logger = LogManager.getLogger();
+    private static final int MAX_UNIPROT_BATCH_QUERY_SIZE = 250;
+
+    /**
+     * Queries the UniProt mapping service through their Java API library. All Uniprot accession IDs are taken from the Panther
+     * homolog files and used to query UniProt for the associated Gene Name. At time of writing (February 2020) this takes about
+     * 3 hours to complete. The end result is a {targetSpecies}_gene_name_mapping.tsv file for each species.
+     * @param speciesKey String - Shortened version of species name (eg: Bos taurus --> btau).
+     * @param releaseNumber String - Used to create a directory where the produced files are stored.
+     * @param pantherHomologMappings Map<String, Set<String>> - Species-specific homolog mappings.
+     * @throws ServiceException - Thrown by UniProtService class if service is unavailable.
+     * @throws InterruptedException - Thrown if the Sleep process that occurs after every batch query is interrupted.
+     * @throws IOException - Thrown if unable to create/write to the mapping files.
+     */
+    public static void retrieveAndStoreGeneNameMappings(String speciesKey, String releaseNumber, Map<String, Set<String>> pantherHomologMappings) throws ServiceException, InterruptedException, IOException {
+
+        // Get all gene names associated with all UniProt identifiers.
+        Set<String> uniprotAccessionsToGeneNames = retrieveGeneNameMappings(pantherHomologMappings);
+        // Store results to {targetSpecies}_gene_name_mapping.tsv file.
+        storeGeneNameMappings(speciesKey, releaseNumber, uniprotAccessionsToGeneNames);
+    }
+
+    /**
+     * Organizes all UniProt identifiers in the pantherMappings object into chunks of 250 identifiers. This is due to a
```

Comment:
There are a few places in the comments starting from here where the size of the chunks are mentioned to be 250.  Maybe you could replace the specific number with something like "chunks of a predetermined size as defined by the MAX_UNIPROT_BATCH_QUERY_SIZE constant" so it refers to where the number is set.  This way if the size changes, you don't have to worry about updating the comments to sync them with the constant.